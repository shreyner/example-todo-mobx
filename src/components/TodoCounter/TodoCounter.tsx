import * as React from 'react';

export interface TodoCounterProps {
  count: number;
}

export function TodoCounter({ count }: TodoCounterProps) {
  return <div>Count: {count}</div>;
}
