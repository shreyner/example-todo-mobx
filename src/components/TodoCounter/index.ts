import { inject, observer } from 'mobx-react';
import { TodoCounter } from './TodoCounter';

export default inject(({todoStore}) => ({
  count: todoStore.countTodos
}))(observer(TodoCounter));
