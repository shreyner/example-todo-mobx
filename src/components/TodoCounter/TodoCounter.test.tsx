import * as React from 'react';
import * as render from 'react-test-renderer';
import { TodoCounter } from './TodoCounter';

test('Render TodoCounter', () => {
  const treeJson = render.create(
    <TodoCounter count={10} />
  ).toJSON();

  expect(treeJson).toMatchSnapshot();
});
