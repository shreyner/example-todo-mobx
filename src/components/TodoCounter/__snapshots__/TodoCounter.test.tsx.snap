// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render TodoCounter 1`] = `
<div>
  Count: 
  10
</div>
`;
